frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts

```diff
@@ -0,0 +1,105 @@
+import { NS } from '../utils/consts';
+import { commonFlows } from './common';
+import { POOL_PROGRESS } from '../../src/constants/storage-pool-const';
+
+// Pool var
+export const poolName: string = 'example.pool';
+export const replicaCount: string = '2';
+export const volumeType: string = 'ssd';
+
+export const poolMessage: {
+  [key in POOL_PROGRESS]?: string;
+} = {
+  [POOL_PROGRESS.FAILED]: `Pool "${poolName}" already exists`,
+  [POOL_PROGRESS.CREATED]: `Pool ${poolName} was successfully created`,
+  [POOL_PROGRESS.NOTALLOWED]:
+    "Pool management tasks are not supported for default pool and Openshift Container Storage's external mode.",
+};
+
+export const navigateToBlockPool = () => {
+  commonFlows.navigateToOCS();
+  cy.byLegacyTestID('horizontal-link-Block Pools').click();
+};
+
+export const populateBlockPoolForm = () => {
+  cy.byTestID('new-pool-name-textbox').type(poolName);
+  cy.byTestID('replica-dropdown').click();
+  cy.byLegacyTestID(replicaCount)
+    .last()
+    .contains(`${replicaCount}-way Replication`)
+    .click();
+  cy.byTestID('volume-type-dropdown').click();
+  cy.byTestID(volumeType).click();
+  cy.byTestID('compression-checkbox').check();
+};
+
+export enum Actions {
+  created = 'created',
+  failed = 'failed',
+  notAllowed = 'notAllowed',
+}
+
+export const blockPoolFooter = (action: string, visible: boolean = false) => {
+  switch (action) {
+    case Actions.failed:
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.FAILED]);
+      cy.byLegacyTestID('modal-try-again-action').should('be.visible');
+      cy.byLegacyTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
+      break;
+    case Actions.created:
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.CREATED]);
+      cy.byLegacyTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
+      break;
+    case Actions.notAllowed:
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.NOTALLOWED]);
+      cy.byLegacyTestID('modal-close-action')
+        .should('be.visible')
+        .click();
```

Comment:
```suggestion
      cy.byLegacyTestID('modal-close-action').click();
```
Commands perform certain validation before they run hence the `visible` assertion is not required.